File: classifier/data/raw/human/samples/project_euler_zig/problem_061.zig
const std = @import("std");

// Project Euler 61: cyclical four-digit polygonal numbers.
const Candidate = struct { value: u16, prefix: u8, suffix: u8 };

fn polygonal(kind: usize, n: u32) u32 {
    return switch (kind) {
        0 => n * (n + 1) / 2,
        1 => n * n,
        2 => n * (3 * n - 1) / 2,
        3 => n * (2 * n - 1),
        4 => n * (5 * n - 3) / 2,
        else => n * (3 * n - 2),
    };
}

fn search(lists: *const [6][100]Candidate, counts: *const [6]usize, chosen: *[6]Candidate, used: [6]bool, depth: usize) bool {
    if (depth == 6) return chosen[5].suffix == chosen[0].prefix;
    for (0..6) |kind| {
        if (used[kind]) continue;
        for (lists[kind][0..counts[kind]]) |candidate| {
            if (depth != 0 and candidate.prefix != chosen[depth - 1].suffix) continue;
            var next_used = used;
            next_used[kind] = true;
            chosen[depth] = candidate;
            if (search(lists, counts, chosen, next_used, depth + 1)) return true;
        }
    }
    return false;
}

pub fn main() !void {
    var lists: [6][100]Candidate = undefined;
    var counts = [6]usize{ 0, 0, 0, 0, 0, 0 };
    for (0..6) |kind| {
        for (1..200) |n| {
            const value = polygonal(kind, @intCast(n));
            if (value < 1000 or value >= 10000) continue;
            const suffix: u8 = @intCast(value % 100);
            if (suffix < 10) continue;
            lists[kind][counts[kind]] = .{ .value = @intCast(value), .prefix = @intCast(value / 100), .suffix = suffix };
            counts[kind] += 1;
        }
    }
    var chosen: [6]Candidate = undefined;
    const found = search(&lists, &counts, &chosen, .{ false, false, false, false, false, false }, 0);
    if (!found) return error.NoSolution;
    var total: u32 = 0;
    for (chosen) |candidate| total += candidate.value;
    var buffer: [32]u8 = undefined;
    const text = try std.fmt.bufPrint(&buffer, "{d}\n", .{total});
    _ = std.c.write(1, text.ptr, text.len);
}

File: classifier/data/raw/ai/samples/project_euler_java/Problem061.java
// Project Euler 61: cyclic four-digit polygonal numbers.
import java.util.*;

public class Problem061 {
    static List<Integer>[] values = new List[6];
    static int[] path = new int[6];
    static boolean[] used = new boolean[6];

    public static void main(String[] args) {
        for (int type = 0; type < 6; type++) {
            values[type] = new ArrayList<>();
            int sides = type + 3;
            for (int n = 1;; n++) {
                int v = n * ((sides - 2) * n - (sides - 4)) / 2;
                if (v >= 10_000) break;
                if (v >= 1_000 && v % 100 >= 10) values[type].add(v);
            }
        }
        for (int first : values[0]) {
            path[0] = first;
            used[0] = true;
            if (search(1)) return;
            used[0] = false;
        }
    }

    static boolean search(int depth) {
        if (depth == 6) {
            if (path[5] % 100 == path[0] / 100) {
                int sum = 0;
                for (int v : path) sum += v;
                System.out.println(sum);
                return true;
            }
            return false;
        }
        int prefix = path[depth - 1] % 100;
        for (int type = 1; type < 6; type++) if (!used[type]) {
            for (int v : values[type]) if (v / 100 == prefix) {
                used[type] = true;
                path[depth] = v;
                if (search(depth + 1)) return true;
                used[type] = false;
            }
        }
        return false;
    }
}
